internal/token/migration_test.go

```diff
@@ -0,0 +1,162 @@
+package token_test
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd-oidc-brokers/internal/token"
+	"golang.org/x/crypto/scrypt"
+)
+
+func TestUseOldEncryptedToken(t *testing.T) {
+	tests := map[string]struct {
+		passwordFileExists          bool
+		oldEncryptedTokenFileExists bool
+
+		expectedRet bool
+		expectError bool
+	}{
+		"Success when password file exists":            {passwordFileExists: true, expectedRet: false},
+		"Success when old encrypted token file exists": {oldEncryptedTokenFileExists: true, expectedRet: true, expectError: false},
+
+		"Error if neither file exists": {expectedRet: false, expectError: true},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			passwordPath := t.TempDir() + "/password"
+			oldEncryptedTokenPath := t.TempDir() + "/oldtoken"
+
+			if tc.passwordFileExists {
+				err := os.WriteFile(passwordPath, []byte("password"), 0600)
+				require.NoError(t, err, "WriteFile should not return an error")
+			}
+			if tc.oldEncryptedTokenFileExists {
+				err := os.WriteFile(oldEncryptedTokenPath, []byte("encryptedtoken"), 0600)
+				require.NoError(t, err, "WriteFile should not return an error")
+			}
+
+			got, err := token.UseOldEncryptedToken(passwordPath, oldEncryptedTokenPath)
+			if tc.expectError {
+				require.Error(t, err, "UseOldEncryptedToken should return an error")
+			} else {
+				require.NoError(t, err, "UseOldEncryptedToken should not return an error")
+				require.Equal(t, tc.expectedRet, got, "UseOldEncryptedToken should return the expected value")
+			}
+		})
+	}
+}
+
+func TestLoadOldEncryptedAuthInfo(t *testing.T) {
+	tests := map[string]struct {
+		setup       func(t *testing.T, path, password string)
+		password    string
+		expectedRet token.AuthCachedInfo
```

Comment:
Missing the whitespace between the input and the expected returns

Reply:
added